apps/docs/docs/vue-query.mdx

```diff
@@ -0,0 +1,176 @@
+import { InstallTabs } from '@site/src/components/InstallTabs';
+
+# Vue Query
+
+## Installation
+
+<InstallTabs packageName="@ts-rest/vue-query @tanstack/vue-query" />
+
+This is a client using `@ts-rest/vue-query`, using `@tanstack/vue-query` under the hood.
+
+## Init TanStack Query Vue plugin
+
+To use the query client in your Vue app, you'll need to initialise the `react-query` client in your app.
+
+```ts
+import { VueQueryPlugin } from '@tanstack/vue-query';
+
+createApp(App).use(VueQueryPlugin).mount('#app');
+```
+
+## initQueryClient
+
+The below snippet is how you'd create a query client, this is pretty much the same structure as the `@ts-rest/core` client.
+
+```tsx
+import { initQueryClient } from '@ts-rest/vue-query';
+
+export const client = initQueryClient(router, {
+  baseUrl: 'http://localhost:3333',
+  baseHeaders: {},
+  api?: () => ... // <- Optional Custom API Fetcher (see below)
+});
+```
+
+To customise the API, follow the same docs as the core client [here](/docs/core/custom)
+
+:::tip
+
+By default, ts-rest encodes query parameters as regular URL encoded strings (with support for nested objects, arrays etc) with full decode compatibility from [`qs`](https://www.npmjs.com/package/qs)
+
+To encode query parameters as JSON, you can use the `jsonQuery` option, please note you'll need to configure your backend to support decoding JSON query parameters.
+
+```tsx
+export const client = initQueryClient(router, {
+  ...,
+  jsonQuery: true
+});
+
+```
+
+:::
+
+## useQuery and useMutation
+
+Once you've created a client using `initQueryClient`, you may traverse the object (in the exact same structure as your contract layout) to find the query or mutation you want to use.
+
+```tsx
+const queryResult = client.posts.get.useQuery(
+  ['posts'], // <- queryKey
+  { params: { id: '1' } }, // <- Query params, Params, Body etc (all typed)
+  { staleTime: 1000 } // <- react-query options (optional)
+);
+```
+
+:::tip
+
+The design philosophy of `@ts-rest/vue-query` is to make it as easy as possible to use `react-query` with `@ts-rest`. This means that the `useQuery` and `useMutation` hooks are as close to the original `react-query` hooks as possible, as such we don't abstract away from the query keys or the options. Only the query function itself!
```

Comment:
should this say `react-query`? Shouldn't it be `vue-query`?

Reply:
I was wondering myself. Technically it should probably be TanStack Query in the docs for both `@ts-rest/react-query` and `@ts-rest/vue-query`. I wanted to keep it consistent without changing the existing docs.